### ISSUE TITLE
'Zero-argument form of "super" call is valid only within a method' false positive when creating method in method

### ISSUE THREAD

**Issue by rly** [microsoft/pyright#6681]
**Describe the bug**

VS Code with pyright now highlights the problem 'Zero-argument form of "super" call is valid only within a method' for the text `super` in the below code. However, I believe this usage is perfectly valid. The application code that the snippet below is based on has run fine for years, and it used to not be highlighted as a problem as of sometime recently.

Related to https://github.com/microsoft/pyright/issues/6093, but instead of a class defined in a method, this is for a method defined in a method.

**Code or Screenshots**

```python
class MySubClass(MyClass):

    def make_constructor(self):
        def _func(self):
            super().__init__()

        return _func
```

**VS Code extension or command-line**
VS Code extension, pyright version 1.1.339

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.340, which I just published. It will be included in a future release of pylance.

**Comment by rly** [comment]
Thanks!
